Mathlib/Algebra/Category/Ring/BinaryCoproducts.lean

```diff
@@ -0,0 +1,91 @@
+/-
+Copyright (c) 2024 Justus Springer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Justus Springer
+-/
+import Mathlib.Algebra.Category.Ring.Basic
+import Mathlib.Algebra.Lie.OfAssociative
+import Mathlib.CategoryTheory.Limits.Shapes.BinaryProducts
+import Mathlib.RingTheory.TensorProduct.Basic
+
+/-!
+# The tensor product is a coproduct in the category of commutative rings
+
+In this file, we show that the tensor product over the integers is a coproduct in the category
+of commutative rings, see `tensorProductColimit`.
+
+-/
+
+universe u
+
+noncomputable section
+
+open CategoryTheory Limits
+open scoped TensorProduct
+
+namespace CommRingCat
+
+variable (A B : CommRingCat.{u})
+
+/-- The tensor product `A ⊗[ℤ] B` forms a cocone for `A` and `B`. -/
+@[simps! pt ι]
+def tensorProductCocone : BinaryCofan A B :=
+BinaryCofan.mk
+  (ofHom (Algebra.TensorProduct.includeLeft (S := ℤ)).toRingHom : A ⟶  of (A ⊗[ℤ] B))
+  (ofHom (Algebra.TensorProduct.includeRight (R := ℤ)).toRingHom : B ⟶  of (A ⊗[ℤ] B))
+
+@[simp]
+theorem tensorProductCocAone_inl : (tensorProductCocone A B).inl =
+  ofHom (Algebra.TensorProduct.includeLeft (S := ℤ)).toRingHom := rfl
+
+@[simp]
+theorem tensorProductCocone_inr : (tensorProductCocone A B).inr =
+  ofHom (Algebra.TensorProduct.includeRight (R := ℤ)).toRingHom := rfl
+
+/-- The tensor product `A ⊗[ℤ] B` is a coproduct for `A` and `B`. -/
+@[simps]
+def tensorProductColimit : IsColimit (tensorProductCocone A B) where
+  desc (s : BinaryCofan A B) :=
+    ofHom (Algebra.TensorProduct.lift s.inl.hom.toIntAlgHom s.inr.hom.toIntAlgHom
+      (fun _ _ => by apply Commute.all)).toRingHom
+  fac (s : BinaryCofan A B) := by
+    rintro ⟨j⟩
+    cases j <;> ext a
+    · simp only [forget_obj, pair_obj_left, tensorProductCocone_pt, Functor.const_obj_obj,
+        BinaryCofan.ι_app_left, pair_obj_right, AlgHom.toRingHom_eq_coe, CommRingCat.comp_apply,
+        coe_of]
+      erw [Algebra.TensorProduct.lift_tmul (hfg := fun _ _ => by apply Commute.all)]
```

Comment:
Do you understand why this `erw` is needed? Is there a missing lemma?
Or maybe it goes away after merging master? Because there were refactors or concrete categories like `CommRingCat` in the past days.

Reply:
No I don't know exactly why it's needed. I'll merge master and look at it again.

Reply:
You're right, simp closes the goal now (it did not originally as far as I recall).